src/libs/Navigation/AppNavigator/createCustomStackNavigator/CustomRouter.ts

```diff
@@ -58,24 +53,25 @@ const addCentralPaneNavigatorRoute = (state) => {
         },
     };
     state.routes.splice(1, 0, centralPaneNavigatorRoute);
-    // eslint-disable-next-line no-param-reassign
-    state.index = state.routes.length - 1;
+    // eslint-disable-next-line no-param-reassign, @typescript-eslint/non-nullable-type-assertion-style
+    (state.index as number) = state.routes.length - 1;
 };
 
-function CustomRouter(options) {
+function CustomRouter(options: ResponsiveStackNavigatorRouterOptions) {
     const stackRouter = StackRouter(options);
 
     return {
         ...stackRouter,
-        getRehydratedState(partialState, {routeNames, routeParamList}) {
+        getRehydratedState(partialState: StackNavigationState<ParamListBase>, {routeNames, routeParamList, routeGetIdList}: RouterConfigOptions): StackNavigationState<ParamListBase> {
             // Make sure that there is at least one CentralPaneNavigator (ReportScreen by default) in the state if this is a wide layout
             if (!isAtLeastOneCentralPaneNavigatorInState(partialState) && !options.getIsSmallScreenWidth()) {
                 // If we added a route we need to make sure that the state.stale is true to generate new key for this route
+
                 // eslint-disable-next-line no-param-reassign
-                partialState.stale = true;
+                (partialState.stale as boolean) = true;
                 addCentralPaneNavigatorRoute(partialState);
             }
-            const state = stackRouter.getRehydratedState(partialState, {routeNames, routeParamList});
+            const state = stackRouter.getRehydratedState(partialState, {routeNames, routeParamList, routeGetIdList});
```

Comment:
Why is `routeGetIdList` being added here?

Reply:
Because `routeGetIdList` is a required param of `RouterConfigOptions`.
`getRehydratedState` [typing](https://github.com/react-navigation/react-navigation/blob/main/packages/routers/src/types.tsx#L171-L174)
`RouterConfigOptions` [typing](https://github.com/react-navigation/react-navigation/blob/main/packages/routers/src/types.tsx#L136)